### ISSUE TITLE
Feature request: Disallow function signature verbosity/ Force signature location

### ISSUE THREAD

**Issue by QwertyZW** [palantir/tslint#3231]
Disallow function declarations with redundant verbosity
```ts
let myVerboseSignature : (a:number) => number = function myVerboseSignature(a : number) : number{
    return a;
}
```

Allow the linter to be configured such that the type is either after the let symbol declaration or as a part of the function declaration, but not both

**Comment by adidahiya** [comment]
Yeah, I like this idea. It could be called `no-redundant-typedef`.

**Comment by ajafff** [comment]
FYI: I'm currently experimenting with an implementation that shows all redundant type annotations.
https://github.com/ajafff/tslint-consistent-codestyle/blob/master/docs/no-unnecessary-type-annotation.md
It's still a work in progress and at this moment only checks parameters and variables, no return types.
Also note that you can not configure at which location you prefer your type annotations, yet.

If you want to give it a try, please report all false positives you may encounter.

Once I think it's stable enough, I'll submit it as PR to this repo.

**Comment by JoshuaKGoldberg** [comment]
Note: per #4534, this issue will be closed in less than a month if no PR is sent to add the rule. If you _really_ need the rule, custom rules are always an option and can be maintained outside this repo!

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for new rules. See #4534. 😱

If you'd like to see this rule implemented, you have two choices:
* **Recommended:** Use [typescript-eslint](https://typescript-eslint.io) to lint your TypeScript code with ESLint ✅
* **Legacy:** Write it as a [custom TSLint rule](https://palantir.github.io/tslint/develop/custom-rules/) 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
